### PR TITLE
fix-NXP-29401-showconf-invalid-json

### PR DESCRIPTION
**Pull request by kevinleturc**
NXP-29401: Fix invalid json when newline is present

### CODE REVIEW

File: server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java [hunk 1]
```diff
@@ -2267,15 +2272,20 @@ protected boolean pkgReset() {
...
-                    PackageInfo.class, ConfigurationInfo.class, KeyValueInfo.class);
-            printXMLOutput(jaxbContext, instance, System.out);
+            printInstanceXMLOutput(instance, System.out);
         } catch (JAXBException | XMLStreamException | FactoryConfigurationError e) {
             throw new NuxeoLauncherException("Output serialization failed: " + e.getMessage(), EXIT_CODE_NOT_RUNNING,
                     e);
         }
     }
 
+    protected void printInstanceXMLOutput(InstanceInfo instance, OutputStream out)
```
Comment (NourNuxeo) [inline server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java:2282]: if it's been extracted for being able to choose the outputStream easily in the test, maybe I would indicate it in a comment. Or is it going to be used and reused?

File: server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java [hunk 2]
```diff
@@ -172,4 +175,27 @@ private void checkParsing(NuxeoLauncher launcher) throws GeneralSecurityExceptio
         launcher.encrypt();
     }
 
+    /*
+     * NXP-29401
+     */
```
Comment (NourNuxeo) [inline server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java:180]: if it's a simple comment shouldn't it be one-lined?

**Review by ataillefer — APPROVED** [review]

File: server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java (hunk 1 above)
Comment (kevinleturc) [reply server/nuxeo-launcher/src/main/java/org/nuxeo/launcher/NuxeoLauncher.java:2282]: In fact, this was present in 10.10 but removed in 11 and now back for test yes. I'm not sure it needs a comment.

File: server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java (hunk 2 above)
Comment (kevinleturc) [reply server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java:180]: I would say it is free. As I like to have room in the code for clarity, I converted it to javadoc.

**Review by ataillefer — APPROVED** [review]

File: server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java (hunk 2 above)
Comment (NourNuxeo) [reply server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java]: but shouldn't it start with two stars to be considered Javadoc?

Comment (kevinleturc) [reply server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java]: You can have a look at the code which reflects the change https://github.com/nuxeo/nuxeo/pull/4281/files#diff-c7467595cf49ff4ffa220ef476aa4384R178

Comment (NourNuxeo) [reply server/nuxeo-launcher/src/test/java/org/nuxeo/launcher/TestNuxeoLauncher.java]: oh sorry, outdated!

**Review by NourNuxeo — APPROVED** [review]

**Pull request merged**